### PR TITLE
feat: Issue NEW-36 Add base entities

### PR DESCRIPTION
**Pull request by SimeonIksanov**

### CODE REVIEW

File: 02. Business/NewHabr.Domain/Models/User.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public ICollection<LikedArticle> LikedArticles { get; set; } = null!;
+
+    public ICollection<LikedComment> LikedComments { get; set; } = null!;
+
+    public ICollection<LikedUser> LikedUsers { get; set; } = null!;
+
+    public ICollection<LikedUser> ReceivedLikes { get; set; } = null!;
+
+
+    public bool Banned { get; set; }
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/User.cs:40]: BanReason

File: 02. Business/NewHabr.Domain/Models/Category.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace NewHabr.Domain.Models;
+
+public class Category : BaseEntity<int>
+{
+    [Required]
+    public string Text { get; set; } = string.Empty;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/Category.cs]: Text => Name

File: 02. Business/NewHabr.Domain/Models/EntityType.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿//namespace NewHabr.Domain.Models;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/EntityType.cs]: На удаление

File: 02. Business/NewHabr.Domain/Models/SecureQuestion.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace NewHabr.Domain.Models;
+
+public class SecureQuestion : BaseEntity<int>
+{
+    public string Question { get; set; } = string.Empty;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/SecureQuestion.cs]: [Required]

File: 02. Business/NewHabr.Domain/Models/Tag.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace NewHabr.Domain.Models;
+
+public class Tag : BaseEntity<int>
+{
+    [Required]
+    public string Text { get; set; } = string.Empty;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/Tag.cs]: Text => Name

File: 02. Business/NewHabr.Domain/Models/User.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public string? FirstName { get; set; }
+
+    [MaxLength(30)]
+    public string? LastName { get; set; }
+
+    [MaxLength(30)]
+    public string? Patronymic { get; set; }
+
+    [Range(0, 120)]
+    public int? Age { get; set; }
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/User.cs]: Age => BDay

File: 02. Business/NewHabr.Domain/Models/User.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    public bool Banned { get; set; }
+
+    public bool Deleted { get; set; }
+
+    public DateTimeOffset? BanExpiratonDate { get; set; }
+
+    public DateTimeOffset? BannedAt { get; set; }
+
+    [Required]
+    public SecureQuestion SecureQuestion { get; set; } = null!;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/User.cs]: Add SequreQuestionId

File: 02. Business/NewHabr.Domain/Models/Article.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+public class Article : BaseEntity<Guid>
+{
+    [Required, MinLength(10), MaxLength(200)]
+    public string Title { get; set; } = string.Empty;
+
+    [Required]
+    public string Content { get; set; } = string.Empty;
+
+    [Required]
+    public User User { get; set; } = null!;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/Article.cs]: UserId

File: 02. Business/NewHabr.Domain/Models/Comment.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace NewHabr.Domain.Models;
+
+public class Comment : BaseEntity<Guid>
+{
+    [Required]
+    public User User { get; set; } = null!;
```
Comment (Narendin) [inline 02. Business/NewHabr.Domain/Models/Comment.cs]: UserId
ArticleId

File: 02. Business/NewHabr.Domain/Models/User.cs (hunk 1 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/User.cs:40]: added

File: 02. Business/NewHabr.Domain/Models/Category.cs (hunk 2 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/Category.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/EntityType.cs (hunk 3 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/EntityType.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/SecureQuestion.cs (hunk 4 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/SecureQuestion.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/Tag.cs (hunk 5 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/Tag.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/User.cs (hunk 6 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/User.cs]: applied

File: 02. Business/NewHabr.Domain/Models/User.cs (hunk 7 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/User.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/Article.cs (hunk 8 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/Article.cs]: fixed

File: 02. Business/NewHabr.Domain/Models/Comment.cs (hunk 9 above)
Comment (SimeonIksanov) [reply 02. Business/NewHabr.Domain/Models/Comment.cs]: fixed

**Pull request merged**